### PR TITLE
Issue/gh #937:  Media Upload not working with query parameter

### PR DESCRIPTION
**Pull request by yahilmadakiya**

### CODE REVIEW

File: app/main/RTMedia.php [hunk 1]
```diff
@@ -1164,7 +1164,7 @@ function enqueue_scripts_styles() {
 			$url_upload = substr( $url, 0, strrpos( $url, $rtmedia_slug ) );
 			$url        = trailingslashit( $url_upload ) . 'upload/';
 		} else {
-			$url = trailingslashit( $url ) . 'upload/';
+			$url = 'upload' . trim( $url, '/' );
```
Comment (milindmore22) [inline app/main/RTMedia.php]: Why prefix `upload` in url ? 
possible output `uploadhttps://yahil.rtdemo.in/?a=sds`

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 2]
```diff
@@ -281,7 +281,7 @@ function bp_after_activity_post_form() {
 			$url_upload                              = implode( '/', $slug_split );
 			$url                                     = trailingslashit( $url_upload ) . 'upload/';
 		} else {
-			$url = trailingslashit( $url ) . 'upload/';
+			$url = 'upload' . trim( $url, '/' );
```
Comment (milindmore22) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Why prefix upload in url ?
possible output `uploadhttps://yahil.rtdemo.in/?a=sds`

File: app/main/controllers/shortcodes/RTMediaGalleryShortcode.php [hunk 3]
```diff
@@ -74,7 +74,12 @@ static function register_scripts() {
 			$url_upload = substr( $url, 0, strrpos( $url, $rtmedia_slug ) );
 			$url        = trailingslashit( $url_upload ) . 'upload/';
 		} else {
-			$url = trailingslashit( $url ) . 'upload/';
+			if ( strstr( $url, '?' ) ) {
+				$url = explode( '?', $url );
+				$url = $url[0] . 'upload?' . trim( $url[1], '/' );
```
Comment (milindmore22) [inline app/main/controllers/shortcodes/RTMediaGalleryShortcode.php]: missing forward slash after upload 
`https://yahil.rtdemo.in/upload?a=sds`
it should be
`https://yahil.rtdemo.in/upload/?a=sds`

**Comment by BhargavBhandari90** [comment]
@milindmore22 All changes are done. Please review.

**Comment by milindmore22** [comment]
@BhargavBhandari90 
I tried following code
```
$url = 'https://yahil.rtdemo.in/upload?a=sds';



if ( strstr( $url, '?' ) ) {
 $url = explode( '?', $url );
 $url = $url[0] . 'upload/?' . trim( $url[1], '/' );
} else {
$url = trailingslashit( $url ) . 'upload/';
 }
echo $url;

function trailingslashit( $string ) {
   return untrailingslashit( $string ) . '/';
}
function untrailingslashit( $string ) {
       return rtrim( $string, '/' );
}
```
output : 
```https://yahil.rtdemo.in/uploadupload/?a=sds```

**Comment by milindmore22** [comment]
I am not sure about this, but check what if upload page has query string

**Comment by BhargavBhandari90** [comment]
@milindmore22 At that point $url will never get a URl without trailing slash. But we can't ignore this. So I have corrected that part.

**Comment by milindmore22** [comment]
@chandrapatel please do a final review.

**Comment by milindmore22** [comment]
@BhargavBhandari90 I recommend to get this tested thoroughly before merging.

File: app/main/RTMedia.php [hunk 4]
```diff
@@ -1158,13 +1158,20 @@ function enqueue_scripts_styles() {
 		$request_uri = rtm_get_server_var( 'REQUEST_URI', 'FILTER_SANITIZE_URL' );
 		$url          = trailingslashit( $request_uri );
 		$rtmedia_slug = '/' . RTMEDIA_MEDIA_SLUG;
-		// check position of media slug from end of the URL
+		// check position of media slug from end of the URL.
```
Comment (pradeep910) [inline app/main/RTMedia.php]: @bhargav - pls convert this into a php function as its being copied 3 files.

File: app/main/RTMedia.php [hunk 5]
```diff
@@ -1158,13 +1158,20 @@ function enqueue_scripts_styles() {
...
+			// split the url upto the last occurrence of media slug.
 			$url_upload = substr( $url, 0, strrpos( $url, $rtmedia_slug ) );
 			$url        = trailingslashit( $url_upload ) . 'upload/';
 		} else {
-			$url = trailingslashit( $url ) . 'upload/';
+
+			// If url contains '?' then put query string at last.
+			if ( strstr( $url, '?' ) ) {
+				$url = explode( '?', $url );
+				$url = trailingslashit( $url[0] ) . 'upload/?' . trim( $url[1], '/' );
```
Comment (pradeep910) [inline app/main/RTMedia.php]: @BhargavBhandari90 Please add sanity check before using Exploded variables $url[0] and $url[1].

**Comment by BhargavBhandari90** [comment]
@pradeep910 Requested change has been done. Please verify.

**Pull request merged**